### ISSUE TITLE
Property setter parameter type-checked against getter return type instead of setter parameter type

### ISSUE THREAD

**Issue by bartfeenstra** [python/mypy#12900]
**Bug Report**
As shown in the code under *To Reproduce*, mypy checks the value that is set on an attribute not against that attribute's setter's parameter type (`Iterable[Member]`), but against the getter's return type (`List[Member]`). While it may not be common for a setter to accept a supertype of the type returned by the getter, or even a completely unrelated and incompatible type, there are valid reasons to do so and mypy should not make the assumption that both types are identical.

**To Reproduce**

```python
from typing import List, Iterable


class Member:
    pass


class Group:
    @property
    def members(self) -> List[Member]:
        pass

    @members.setter
    def members(self, members: Iterable[Member]) -> None:
        pass


group = Group()
group.members = {Member(), Member()}

# mypy error: test.py:19: error: Incompatible types in assignment (expression has type "Set[Member]", variable has type "List[Member]")
```

**Expected Behavior**

No errors raised by mypy at all.

**Actual Behavior**

mypy erraneously flags the type of the value set on the attribute as incompatible.

**Your Environment**

- Mypy version used: 0.950
- Mypy command-line flags: N/A
- Mypy configuration options from `mypy.ini` (and other config files):
  ```
  check_untyped_defs = True
  no_implicit_optional = True
  ```
- Python version used: 3.9
- Operating system and version: Ubuntu 21.10

**Comment by thomkeh** [comment]
Duplicate of #3004 I think. There was a PR to implement it here: #11643 but it wasn't merged.

**Comment by bartfeenstra** [comment]
Aaah, I couldn't find anything earlier. Thanks for digging that up and linking it :+1: I'll close this issue in favor of https://github.com/python/mypy/issues/3004.
